druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -278,12 +282,24 @@ public static DruidQuery extendQuery(DruidQuery query,
       if (!TimeExtractionFunction.isValidTimeExtract((RexCall) rexNode)) {
         return Pair.of(null, null);
       }
-      extractionFunction =
-          TimeExtractionFunction.createExtractFromGranularity(granularity,
-              druidQuery.getConnectionConfig().timeZone());
-      columnName =
-          extractColumnName(((RexCall) rexNode).getOperands().get(1), rowType, druidQuery);
-
+      RexNode extractValueNode = ((RexCall) rexNode).getOperands().get(1);
+      if (extractValueNode.getType().getSqlTypeName() == SqlTypeName.DATE
+          || extractValueNode.getType().getSqlTypeName() == SqlTypeName.TIMESTAMP) {
+        // Use 'UTC' at the extraction level
+        extractionFunction =
+            TimeExtractionFunction.createExtractFromGranularity(
+                granularity, DateTimeUtils.UTC_ZONE.getID());
+        columnName = extractColumnName(extractValueNode, rowType, druidQuery);
+      } else if (extractValueNode.getType().getSqlTypeName()
```

Comment:
seems like this check that the type of rexNode is ```TIMESTAMP_WITH_LOCAL_TIME_ZONE``` is done multiple times i think we can extract this as a method and use that instead something like ```boolean isTimestampWLTZ(RexNode node)```